app/extend_block.go

```diff
@@ -0,0 +1,27 @@
+package app
+
+import (
+	"github.com/celestiaorg/celestia-app/pkg/da"
+	"github.com/celestiaorg/celestia-app/pkg/shares"
+	"github.com/celestiaorg/celestia-app/pkg/square"
+	"github.com/celestiaorg/rsmt2d"
+	coretypes "github.com/tendermint/tendermint/types"
+)
+
+// ExtendBlock extends the given block data into a data square for a given app
+// version.
+func ExtendBlock(data coretypes.Data, appVersion uint64) (*rsmt2d.ExtendedDataSquare, error) {
```

Comment:
this is just going off of what was requested by celestia-node in https://github.com/celestiaorg/celestia-app/issues/1952, so the same api of that function was used

Reply:
Do we want to add a check then that the square size calculated equals the square size provided in Data?

Reply:
actually after thinking about this more, I don't think we should add this check unless we are commiting to the value as it makes it easy to make the celestia-node to reject an otherwise valid block simply by modifying an uncommitted value